[ci skip] clarify the benefits of using ActiveJob's execution wrappers (#49301)
rails/rails: guides/source/active_job_basics.md

GuestsCleanupJob.perform_later(guest1, guest2, filter: 'some_filter')
```

> NOTE: Jobs are meant to be run with the `.perform_later` and `.perform_now`

Reviewer: ```suggestion
NOTE: Jobs are meant to be run with the `perform_later` and `perform_now`
```
You can directly use the method name without adding the period.

> wrappers: running the job's `.perform` method directly is possible but removes

Reviewer: ```suggestion
wrappers: running the job's `perform` method directly is possible but removes
```

GuestsCleanupJob.perform_later(guest1, guest2, filter: 'some_filter')
```

NOTE: Jobs are meant to be run with the `perform_later` and

> `perform_now` wrappers: running the job's `perform` method directly is

Reviewer 2: This is a pretty long sentence. What do you think about splitting it with a period instead?
```suggestion
`perform_now` wrappers. Running the job's `perform` method directly is
```

possible but removes the benefit of using a queue and skips all the

> logic provided by the ActiveJob wrappers like callbacks or exception

Reviewer 2: As `ActiveJob` in this case references the component, not the Ruby module, it should have a space.
See also: https://edgeguides.rubyonrails.org/api_documentation_guidelines.html#wording

```suggestion
logic provided by the Active Job wrappers like callbacks or exception
```
